daemon/daemon.go

```diff
@@ -123,6 +123,9 @@ type Daemon struct {
 	// Used to synchronize generation of daemon's BPF programs and endpoint BPF
 	// programs.
 	compilationMutex *lock.RWMutex
+
+	// ipcacheListeners lists all parties interested in IP -> ID mappings.
+	ipcacheListeners []ipcache.IPIdentityMappingListener
```

Comment:
This is a little out of scope for this PR but could we move this to the ipcache package? It's a singleton that's private to that package except we want to notify the listeners manually. We can have a function to do just that as part of the package (as weird as I find the whole thing).
One day, when someone wants to add a new listener they won't realise that this is a constant and they will change it.

Reply:
Sure. #3976

Reply:
Thanks :)